pandas/core/frame.py

```diff
@@ -4610,33 +4609,19 @@ def _maybe_casted_values(index, labels=None):
 
             # if we have the labels, extract the values with a mask
             if labels is not None:
-                mask = labels == -1
-
-                # we can have situations where the whole mask is -1,
-                # meaning there is nothing found in labels, so make all nan's
-                if mask.all():
-                    values = np.empty(len(mask))
-                    values.fill(np.nan)
+                if isinstance(values, np.ndarray):
```

Comment:
can you move _maybe_casted_values (and call it maybe_casted_values) to pandas.core.dtypes.cast.

This looks a bit duplicative of some other code there, but we can refactor that after.